### DISCUSSION TITLE
Rest API endpoint to fetch logs for a pipelinerun

### DISCUSSION THREAD

**Discussion by avnipurna** [tektoncd/pipeline#4324] [Q&A] [answered]
Hi, 
I'm currently working on integrating Tekton with our current app. I was looking to fetch logs for the pipelinerun via a rest api instead of individually fetching all the  pods corresponding to each task and pulling the logs for each. 

Is there any rest api to do exactly what the tekton cli does - 
tkn pipelinerun logs -L  tekton-pipelinerun-test-task-8v826

**Comment by vdemeester** [comment, marked as answer]
Hi @avnipurna, there is not such rest api as, for now, tekton/pipeline is just a set of CRDs and a controller/webhook.

**Comment by sanjiv-wesco** [comment]
REST / HTTP Api developed yet or still not developed?.

**Reply by khrm** [reply]
You can try using results.

**Comment by khrm** [comment]
[Results](https://github.com/tektoncd/results) provides GRPC/Rest API calls to fetch logs.
